### PR TITLE
Dmd more updates

### PR DESCRIPTION
**Pull request by kevinhkhuynh**
Add visualization to DMD examples.

### CODE REVIEW

File: lib/algo/DMD.h [hunk 1]
```diff
@@ -41,6 +41,14 @@ class DMD
      */
     DMD(int dim, double dt);
 
+    /**
+     * @brief Constructor.
+     *
+     * @param[in] base_file_name The base part of the file of the
```
Comment (dylan-copeland) [inline lib/algo/DMD.h:47]: file -> filename

File: lib/algo/DMD.h [hunk 2]
```diff
@@ -102,6 +110,22 @@ class DMD
      */
     const Matrix* getSnapshotMatrix();
 
+    /**
+     * @brief Load the object state from a file.
+     *
+     * @param[in] base_file_name The base part of the file to load the
```
Comment (dylan-copeland) [inline lib/algo/DMD.h:116]: file -> filename

File: lib/algo/DMD.h [hunk 3]
```diff
@@ -102,6 +110,22 @@ class DMD
...
+     *                           database from.
+     */
+    virtual void load(std::string base_file_name);
+
+    /**
+     * @brief Save the object state to a file.
+     *
+     * @param[in] base_file_name The base part of the file to save the
```
Comment (dylan-copeland) [inline lib/algo/DMD.h:124]: file -> filename

**Review by dylan-copeland — APPROVED** [review]

File: examples/dmd/dg_euler.cpp [hunk 4]
```diff
@@ -480,47 +483,112 @@ int main(int argc, char *argv[])
...
+                dmd_visit_dc.SetCycle(i);
+                dmd_visit_dc.SetTime(ts[i]);
+                dmd_visit_dc.Save();
+            }
+        }
+
+        delete result_dens;
+        delete result_x_mom;
+        delete result_y_mom;
+        delete result_e;
```
[the remark this answers is not in the document]
Comment (kevinhkhuynh) [reply examples/dmd/dg_euler.cpp:559]: I think so, since each call to DMD->predict does a new inside the DMD function.

**Review by siuwuncheung — APPROVED** [review]

**Review by chldkdtn — APPROVED** [review]

**Pull request merged**